simple_rl/agents/func_approx/dsc/OptionClass.py

```diff
@@ -260,7 +262,7 @@ def construct_image_feature_matrix(examples):
 
 	def construct_negative_data_for_one_class_classifier(self):
 		""" Randomly sample states from the global solver's replay buffer and assume they are negative examples. """
-		states = [np.array(experience.state)[-1, :, :] for experience in self.global_solver.replay_buffer.memory]
+		states = [np.array(experience[0])[-1, :, :] for experience in self.global_solver.replay_buffer.memory]
 		num_positive_examples = len(list(itertools.chain.from_iterable(self.positive_examples)))
 		sampled_states = random.sample(states, k=num_positive_examples)
```

Comment:
Note for the future: we dont want to sample the same number of negatives as positives. We want to add a scaling term to the loss function instead so as to not lose the correct topology of the initiation set classifier.